### ISSUE TITLE
Reset origins for pallet asset-index and pallet saft-registry

### ISSUE THREAD

**Issue by clearloop** [ChainSafe/PINT#318]
## Issue summary

Most of the calls inside these two pallets requires `caller` as parameter, the current implementation of `GovernanceOrigin` actually Either<(), CommitteeOrigin> which could not return `AccountId` when executing extrinsic with `Root` origin.

mb we can just use `CommitteeOrigin` for these two pallets, or trying to construct some new origins like `Either<Sudo, CommitteeOrigin>`?

## Other information and links

-

**Comment by mattsse** [comment]
Can you pinpoint me to the calls,
most dispatchables will require a sender because that's the one that will the issuer of the governance proposal.

Is this about the benchmarks that are using `successfull_origin`? 

What's the distinction between both of them `Either<CouncilOrigin, CommitteeOrigin>` didn't we rename them to `GovernanceOrigin `

Also `EnsureOneOf ` can be nested I believe like

```rust
type AuctionInitiate = EnsureOneOf<
	AccountId,
	EnsureRoot<AccountId>,
	EnsureOneOf<Account, ...>,
>;
```

**Comment by clearloop** [comment]
> Is this about the benchmarks that are using successfull_origin?

Yep, for example, if we use `EnsureOneOf`, the `ensure_signed` will come into error because the unwrap of `Root` will throw `Bad Origin` directly

For `EitherOrigin<CouncilOrigin, CommitteeOrigin>`, I'm thinking of a list of AccountId that allow to execute these calls directly

**Comment by mattsse** [comment]
So the `successfull_origin` for `EnsureApprovedByCommittee` is the default address
but `EnsureOneOf<Root, EnsureApprovedByCommittee>` will use `Root::successfull_origin`.
What if we flip it to `EnsureOneOf<EnsureApprovedByCommittee, Root>`?

**Comment by clearloop** [comment]
> So the `successfull_origin` for `EnsureApprovedByCommittee` is the default address
> but `EnsureOneOf<Root, EnsureApprovedByCommittee>` will use `Root::successfull_origin`.
> What if we flip it to `EnsureOneOf<EnsureApprovedByCommittee, Root>`?

Not sure what it use, but Bad Origin occurs, changed to CommitteeOrigin and got the same error, let me find the root case

**Comment by mattsse** [comment]
This most likely occurs because of `ensure_signed(Root)`

and since `EnsureOneOf<Account, L, R>`

```
fn successful_origin() -> O {
	L::successful_origin()
}
```
 `EnsureOneOf<Account, Root, Committee>::successful_origin()` will always return `Root`

**Comment by clearloop** [comment]
> This most likely occurs because either `ensure_signed(Root)`,
> 
> and since `EnsureOneOf<Account, L, R>`
> 
> ```
> fn successful_origin() -> O {
> 	L::successful_origin()
> }
> ```
> 
> `EnsureOneOf<Account, Root, Committee>::successful_origin()` will always return `Root`
> 
> or because the way EnsureApprovedByCommittee handles it's successfull origin type, check the `try_origin`
> `votes.is_accepted(T::MinCouncilVotes::get()).map_err(|_| o)?;`
> we'd need to add a successful Voteaggregate in `EnsureApprovedByCommittee::successful_origin`, instead of
> `CommitteeOrigin::ApprovedByCommittee(Default::default(), Default::default())`
> a successful vote `CommitteeOrigin::ApprovedByCommittee(Default::default(), Voteagregate::successfull)`

Yep, refactoring the `successful_origin` rn
